website/static/js/pages/login-page.js

```diff
@@ -20,6 +23,45 @@ var activateToggleBox = function () {
     }
 };
 
+
+var InstitutionViewModel = function() {
+    var self = this;
+    self.inst_names = ko.observableArray([]);
+    self.insts = {};
+    $osf.ajaxJSON(
+        'GET',
+        window.contextVars.apiV2Prefix + 'institutions/',
+        {
+            isCors: true
+        }
+    ).done( function(response){
+        for (var i = 0; i < response.data.length; i++){
+            var name = response.data[i].attributes.name;
+            self.inst_names.push(name);
+            self.insts[name] = response.data[i].attributes.auth_url;
+        }
+    }).fail(function(response){
+    });
+
+    self.instLogin = function(){
+        var key = $('#selected_inst').val();
+        window.location = self.insts[key];
+    };
+};
+
+
+$(document).ready(function() {
+    var self = this;
+    self.campaign = $('#campaign').val();
```

Comment:
The mix of jQuery, Knockout, and mako (https://github.com/CenterForOpenScience/osf.io/pull/5017/files#diff-6aa4ed71920a03b1327bfba3519f9ca0R45) is kludgy. Can you consolidate these and use data-binding instead?